src/controllers/item.js

```diff
@@ -139,4 +139,33 @@ module.exports = {
       res.status(422).json({ message: err.message ?? err });
     }
   },
+
+  borrowItem: async (req, res) => {
```

Comment:
What is the difference between this controller and the one above `donateItem`? The logic looks almost exactly the same.

Reply:
In the `donateItem` endpoint, the donator will go through the requests and pick one and make the donation from his already donated items. But this endpoint is for the other users to borrow an item that has been posted for donation.

Reply:
Understood! In that case, should the `donateItem` function check in the requests collection and match the item in the request to the items the donator owns? Or maybe this can be handled on the frontend side itself?

Reply:
Exactly, this functionality has to be implemented on either side. Unfortunately, we didn't notice this when we were building the API until the frontend team pointed this out to me. I don't know if the frontend can implement it or not.

Reply:
Alright, so frontend team could do something like this:
1. Get details of the request using GET route `/api/requests/:id`.
2. Get all items `/api/global/all-items` and filter by donator as owner. (Filtering part can be done in frontend code)
3. Match the requested item in the filtered items. (This means check if the donator has the requested item available for donation)
4. If matching item is found, then frontend proceeds to make API call to the donate item route with the matching item ID.

You can suggest this to the frontend team, but if it is not possible on their end then you can setup this same logic in the `donateItem` controller.

What do you think?

Reply:
Thank you very much for the suggestion, it looks great. Just one small question, according to what should we make the check if the items match or not? Because names might be a little bit different. Should they check if it matches one or two words from the name and to match the same category, is this ok?

Reply:
That is a good point. Yes, checking for one or two matching words and category should be fine for now.
You can even recommend the frontend team to ensure they set up the data in such a way that matches will exist so that they can have a smooth demo on graduation day.

Reply:
Thank you very much @Shrreya for your time 😊 
I will try to enter one of the front-end team's standups after it gets merged to give them a better understanding of how to implement it, or if I don't find the time, I will just send them what you wrote here 😅 😀

Reply:
You're always welcome and sounds good!
I'm marking this PR approved now, but do check in with @Ammar-64 for his review as well.
Don't mark this conversation as resolved so that he gets a chance to go through this discussion.

Reply:
Oh, and on second thought, the matching part can be done manually by the donator too. 😅
The donator can look at his/her list of available items and click on the matching item. Easy peasy, leave it on the user. 🙈

Reply:
Yeah, that is what was going to be implemented in the beginning and that is why I created the second endpoint. When a user wants to make a donation to the request, they will be prompted with their available items and pick on from them. 
I think this is the easiest way and as an extra check they can make filtering by type to make the likely hood of a user donating the wrong item smaller.

Reply:
Yup, filtering would be a good improvement on the functionality.

So basically there are 2 types of donation:
1. Borrower makes a request and then donator donates item by looking at requests.
2. Donator posts items available for donation and then borrower directly borrows an item.

Reply:
Exactly. I might just send them the link to this discussion to clarify how the endpoints should work 😅.

Reply:
Haha okay, thanks for the prompt responses as always @AmmarBaki2 
Long discussion over for now. 😅